cabal.project

```diff
@@ -4,3 +4,13 @@ index-state: 2023-01-11T00:00:00Z
 packages:
   cardano-prelude
   cardano-prelude-test
+
+allow-newer:
+  , *:base
+  , *:ghc-prim
+  , *:template-haskell
+  , hedgehog:*
+  , protolude:binary
+  , protolude:bytestring
+  , protolude:text
+  , th-compat:*
```

Comment:
I believe we'd want to guard this being some 9.6 compat cabal.project.local for now?
We do something similar with custom cabal.project.locals in cardano-node.

Reply:
Done!